brightener.h

```diff
@@ -1,17 +1,33 @@
-#pragma once
-#include <cstdint>
-
-struct Image {
-	int rows;
-	int columns;
-	uint8_t pixels[1024 * 1024]; // max 1k x 1k image
-};
-
-class ImageBrightener {
-private:
-	Image m_inputImage;
-public:
-	ImageBrightener(Image& inputImage);
-	int BrightenWholeImage();
-	Image GetImage();
-};
+#ifndef BRIGHTENER_H
+#define BRIGHTENER_H
+
+#include <vector>
+#include <memory>
+
+class Image {
+public:
+    int rows;
+    int columns;
+    std::vector<int> pixels;
+
+    Image(int r, int c, int initialBrightness = 0)
+        : rows(r), columns(c), pixels(r* c, initialBrightness) {}
+
+    int& pixel(int x, int y) {
+        return pixels[x * columns + y];
+    }
+};
+
+class ImageBrightener {
+public:
+    explicit ImageBrightener(std::unique_ptr<Image> inputImage);
+    int BrightenWholeImage();
+    const Image& GetImage() const;

```

Comment:
Good approach to return a const reference to the `Image` data member. However, if I have to access the pixels in this image, the `pixel` function is not const, so I cannot invoke it. How do I get the pixels of the image after brightening?